python/plugins/processing/algs/qgis/Grid.py

```diff
@@ -147,27 +151,134 @@ def processAlgorithm(self, parameters, context, feedback):
 
+        if idx == 0:
+            outputWkb = QgsWkbTypes.Point
+        elif idx == 1:
+            outputWkb = QgsWkbTypes.LineString
+        else:
+            outputWkb = QgsWkbTypes.Polygon
         (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT, context,
-                                               fields, QgsWkbTypes.Polygon, crs)
+                                               fields, outputWkb, crs)
 
         if idx == 0:
-            self._rectangleGrid(
-                sink, width, height, originX, originY, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
+            self._pointGrid(
+                sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
         elif idx == 1:
-            self._diamondGrid(
-                sink, width, height, originX, originY, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
+            self._lineGrid(
+                sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
         elif idx == 2:
+            self._rectangleGrid(
+                sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
+        elif idx == 3:
+            self._diamondGrid(
+                sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
+        elif idx == 4:
             self._hexagonGrid(
-                sink, width, height, originX, originY, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
+                sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback)
 
         return {self.OUTPUT: dest_id}
 
-    def _rectangleGrid(self, sink, width, height, originX, originY,
-                       hSpacing, vSpacing, hOverlay, vOverlay, feedback):
-        ft = QgsFeature()
 
-        columns = int(math.ceil(float(width) / (hSpacing - hOverlay)))
-        rows = int(math.ceil(float(height) / (vSpacing - vOverlay)))
+    def _pointGrid(self, sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback):
+        feat = QgsFeature()
+
+        columns = int(math.ceil(float(bbox.width()) / (hSpacing - hOverlay)))
+        rows = int(math.ceil(float(bbox.height()) / (vSpacing - vOverlay)))
+
+        cells = rows * columns
+        count_update = cells * 0.05
+
+        id = 1
+        count = 0
+
+        for col in range(columns):
+            for row in range(rows):
+                x = bbox.xMinimum() + (col * hSpacing - col * hOverlay)
+                y = bbox.yMaximum() - (row * vSpacing - row * vOverlay)
+                feat.setGeometry(QgsGeometry.fromPointXY(QgsPointXY(x,y)))
+                feat.setAttributes([x, y, x + hSpacing, y + vSpacing, id])
+                sink.addFeature(feat, QgsFeatureSink.FastInsert)
+
+                id += 1
+                count += 1
+                if int(math.fmod(count, count_update)) == 0:
+                    feedback.setProgress(int(count / cells * 100))
+
+    def _lineGrid(self, sink, bbox, hSpacing, vSpacing, hOverlay, vOverlay, feedback):
```

Comment:
i'm guessing this is directly copied from the removed alg?

Reply:
Yes, no change applied to the function within that def, straight forward copy/paste.